### PR TITLE
removed the function for getURLParam and replaced with URLSearchParam method

### LINKED ISSUE #9765: Replace getURLParam with URLSearchParams - JavaScript clean up
### Is your proposal related to a problem?

Currently we have code that uses regex parse the URL location to get the query param (e.g. `q=foo`).

https://github.com/wagtail/wagtail/blob/f31cce6786fbe2a385367a19a16bcdd6d014a5f1/client/src/entrypoints/admin/core.js#L308-L317

### Describe the solution you'd like

* Replace this usage with the URLSearchParams approach used elsewhere in the code - https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get

Essentially, copy this code

https://github.com/wagtail/wagtail/blob/f31cce6786fbe2a385367a19a16bcdd6d014a5f1/client/src/entrypoints/admin/core.js#L281-L282

### Describe alternatives you've considered

* This is a small bit of clean up work that avoids complex functions when the browser now supports this with native APIs, recommend we just get this done.

### Additional context

* This is part of prep work for the Stimulus project, cleaning up things before we pull them out to Stimulus controllers

### PR DESCRIPTION
**Pull request by Lovelyfin00**
Fixes #9765 

## Fix Summary

-  Removed the getURLParam function and replaced it with the URLSearchParams method.
- The function was called in a new variable (currentQuery) **line 272**, which I removed and replaced with the get method of the URLSearchParams.
- On running `npm run lint` line 307 gave me errors which I assume is because of this eslint rule `    // eslint-disable-next-line func-names`
- Since `newQuery`  local variable in line 270 was used in `searchParams.set()`, I had to remove it from that line and move it to line 267.

### CODE REVIEW

File: client/src/entrypoints/admin/core.js [hunk 1]
```diff
@@ -305,15 +305,8 @@ $(() => {
...
-      const results = new RegExp('[\\?&]' + name + '=([^]*)').exec(
-        window.location.search,
-      );
-      if (results) {
-        return results[1];
-      }
-      return '';
-    };
+    const searchParams = new URLSearchParams(window.location.search);
+    searchParams.set('q', newQuery);
```
Comment (lb-) [inline client/src/entrypoints/admin/core.js:309]: I think there may be some confusion here. 

You should probably move the new URL... Thing up to line 270 - inside the function body of `search`

Also I do not see a need to set the query here at all - that's only needed if you intend to update the URL

File: client/src/entrypoints/admin/core.js [hunk 2]
```diff
@@ -264,12 +264,12 @@ $(() => {
 
     // auto focus on search box
     $input.trigger('focus');
+    const newQuery = $input.val();
```
Comment (lb-) [inline client/src/entrypoints/admin/core.js:267]: I don't think this should have moved, pulling this out of the search function will likely break the header search. 

This code reads the search field value, which I assume we would only want to do when we call the `search` function.

**Review by lb- — CHANGES_REQUESTED** [review]
Great start on this - thanks for getting a PR up. 

The code doesn't look right as is - I have put a few comments. 

Reach out on Slack if this doesn't make sense. 

Also - when changing code - be sure you validate the behaviour before and after. Maybe add console.log to the search function when working locally to understand what this code is doing and when it's called. 

My concern is that this change as is may not work correctly.

File: client/src/entrypoints/admin/core.js [hunk 3]
```diff
@@ -264,12 +264,12 @@ $(() => {
...
 
     // eslint-disable-next-line func-names
     const search = function () {
-      const newQuery = $input.val();
       // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      const currentQuery = getURLParam('q');
+      const currentQuery = searchParams.get('q');
```
Comment (lb-) [inline client/src/entrypoints/admin/core.js:272]: If you replace these first three lines with something like

```
      const newQuery = $input.val();
      const searchParams = new URLSearchParams(window.location.search);
      const currentQuery = URLSearchParams.get('q');
```

then remove `const searchParams = new URLSearchParams(window.location.search);` from below (line 281)

You should be pretty close I think.

Comment (Lovelyfin00) [reply client/src/entrypoints/admin/core.js:272]: This should work. Since the value for q is already been set. Let me test and push the changes

**Comment by Lovelyfin00** [comment]
I've made the changes as requested.
I tested on C:\Users\HP\Desktop\Wagtail New Codebase\wagtail\wagtail\images\templates\wagtailimages\images\index.html

image

**Review by lb- — APPROVED** [review]
Working great! thanks @Lovelyfin00

**Pull request merged**